gelato/index.ts

```diff
@@ -1,8 +1,40 @@
-import { syncFantomVaultHarvesterTasks } from "./syncFantomVaultHarvesterTasks";
-
-const run = async () => {
-    console.log('>>>>>', 'syncFantomVaultHarvesterTasks()');
-    await syncFantomVaultHarvesterTasks();
-  };
-  
-run();
\ No newline at end of file
+import 'dotenv/config'
+import { ethers, Wallet } from 'ethers';
+import { NonceManage } from '../utility/NonceManage';
+import { TaskSyncer } from './taskSyncer';
+import { IChain, IChainOch, IChains } from './interfaces';
+import { logger, Logger } from '../utility/Logger';
+
+
+//logger.setLevel( 'NonceManage', Logger.levels.DEBUG);
+//logger.setLevel( 'GelatoClient', Logger.levels.DEBUG);
+
+const run = async () : Promise< void> => {
+  const pk = process.env.GELATO_ADMIN_PK!;
+
+  //Object.values( <Readonly< IChains>> require( '../data/chains.js')).forEach( (chain: IChain | IChainOch) =>  {
+  await Promise.all( Object.values( <Readonly< IChains>> require( 
+                                          '../data/chains.js')).map( 
+                                          async (chain: IChain | IChainOch) => {
+    if (!chain.hasOnChainHarvesting)
+      return;
+    if (!( (test: IChainOch | IChain) : test is IChainOch => 'string' === 
```

Comment:
This is a TypeScript "type-guard" which gets the TypeScript interpreter to narrow the type of the `chain` variable to the more precise type. Without it, the TypeScript interpreter will properly complain about typing infractions downstream. Type-guards are a safer and more appropriate pattern for narrowing than forcing it with outright assertions.

Reply:
Can we please move the type guard(s) to a separate function(s) rather than defining inline? Would be so much easier to read.
e.g.
```ts
if ( !isChainOch(chain) ) {
  logger.error( `On-chain harvesting misconfigured for ${chain.id.toUpperCase()}`);
  return;
}
```
elsewhere...
```ts
function isChainOch(chain: IChain | IChainOch): chain is IChainOch {
  return 'string' === typeof chain.ochHarvester 
  && !!chain.ochHarvester.length 
  && 'string' === typeof chain.ochOperations
  && !!chain.ochOperations.length;
}
```

Reply:
> Can we please move the type guard(s) to a separate function(s) rather than defining inline? Would be so much easier to read. 

For sure reasonable if it helps you/others read/review the code. Will do.